src/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEventArgs.cs

```diff
@@ -137,11 +138,42 @@ public IList<ArraySegment<byte>> BufferList
                 StartConfiguring();
                 try
                 {
-                    if (value != null && _buffer != null)
+                    if (value != null)
                     {
-                        throw new ArgumentException(SR.Format(SR.net_ambiguousbuffers, "Buffer"));
+                        if (_buffer != null)
+                        {
+                            // Can't have both set
+                            throw new ArgumentException(SR.Format(SR.net_ambiguousbuffers, "Buffer"));
+                        }
+
+                        // Copy the user-provided list into our internal buffer list,
+                        // so that we are not affected by subsequent changes to the list.
+                        // We reuse the existing list so that we can avoid reallocation when possible.
+                        if (_bufferListInternal == null)
+                        {
+                            _bufferListInternal = new List<ArraySegment<byte>>(value.Count);
+                        }
+                        else
+                        {
+                            _bufferListInternal.Capacity = value.Count;
```

Comment:
Yeah, that's not what I want.

Is there a way to ensure the capacity of the list?  I'm trying to avoid a scenario where we resize on the fly and throw due to OOM, or keep resizing over and over again.